File: e2e/java/src/main/java/io/trzcinski/test/pet/PetClient.java
package io.trzcinski.test.pet;

import org.springframework.web.bind.annotation.*;
import java.util.List;
import java.util.Map;
import org.springframework.lang.Nullable;

import io.trzcinski.test.pet.dto.Pet;
import io.trzcinski.test.commons.dto.ApiResponse;

interface PetClient {

    @PostMapping("/pet/{petId}/uploadImage")
    ApiResponse uploadFile(
        @RequestParam("petId") Integer petId
    );

    @PutMapping("/pet")
    void updatePet(
        @RequestBody() Pet payload
    );

    @PostMapping("/pet")
    void addPet(
        @RequestBody() Pet payload
    );

    @GetMapping("/pet/findByStatus")
    List<Pet> findPetsByStatus(
        @RequestParam("status") List<String> status
    );

    @GetMapping("/pet/findByTags")
    List<Pet> findPetsByTags(
        @RequestParam("tags") List<String> tags
    );

    @GetMapping("/pet/{petId}")
    Pet getPetById(
        @RequestParam("petId") Integer petId
    );

    @PostMapping("/pet/{petId}")
    void updatePetWithForm(
        @RequestParam("petId") Integer petId
    );

    @DeleteMapping("/pet/{petId}")
    void deletePet(
        @RequestHeader("api_key") String api_key,
        @RequestParam("petId") Integer petId
    );
}

File: test-samples/kotlin-retrofit/src/main/kotlin/io/trzcinski/test/pet/PetClient.kt
package out.io.trzcinski.test.pet

import retrofit2.http.*
import retrofit2.Call

import out.io.trzcinski.test.pet.dto.Pet
import out.io.trzcinski.test.commons.dto.ApiResponse

interface PetClient {

    @POST("/pet/{petId}/uploadImage")
    fun uploadFile(
        @Path("petId") petId: Int
    ): Call<ApiResponse>

    @PUT("/pet")
    fun updatePet(
    ): Call<Void>

    @POST("/pet")
    fun addPet(
    ): Call<Void>

    @GET("/pet/findByStatus")
    fun findPetsByStatus(
        @Query("status") status: List<String>
    ): Call<List<Pet>>

    @GET("/pet/findByTags")
    fun findPetsByTags(
        @Query("tags") tags: List<String>
    ): Call<List<Pet>>

    @GET("/pet/{petId}")
    fun getPetById(
        @Path("petId") petId: Int
    ): Call<Pet>

    @POST("/pet/{petId}")
    fun updatePetWithForm(
        @Path("petId") petId: Int
    ): Call<Void>

    @DELETE("/pet/{petId}")
    fun deletePet(
        @Header("api_key") api_key: String,
        @Path("petId") petId: Int
    ): Call<Void>
}